benchmarks/graphics_pipeline/GraphicsBenchmarkApp.cpp

```diff
@@ -526,37 +535,67 @@ void GraphicsBenchmarkApp::SetupSpheresPipelines()
-        for (size_t j = 0; j < kAvailablePsShaders.size(); j++) {
-            for (size_t k = 0; k < kAvailableVbFormats.size(); k++) {
-                // Interleaved pipeline
-                grfx::GraphicsPipelineCreateInfo2 gpCreateInfo  = {};
-                gpCreateInfo.VS                                 = {mVsShaders[i].Get(), "vsmain"};
-                gpCreateInfo.PS                                 = {mPsShaders[j].Get(), "psmain"};
-                gpCreateInfo.vertexInputState.bindingCount      = 1;
-                gpCreateInfo.vertexInputState.bindings[0]       = mSphereMeshes[2 * k + 0]->GetDerivedVertexBindings()[0];
-                gpCreateInfo.topology                           = grfx::PRIMITIVE_TOPOLOGY_TRIANGLE_LIST;
-                gpCreateInfo.polygonMode                        = grfx::POLYGON_MODE_FILL;
-                gpCreateInfo.cullMode                           = grfx::CULL_MODE_BACK;
-                gpCreateInfo.frontFace                          = grfx::FRONT_FACE_CCW;
-                gpCreateInfo.depthReadEnable                    = pDepthTestWrite->GetValue();
-                gpCreateInfo.depthWriteEnable                   = pDepthTestWrite->GetValue();
-                gpCreateInfo.blendModes[0]                      = pAlphaBlend->GetValue() ? grfx::BLEND_MODE_ALPHA : grfx::BLEND_MODE_NONE;
-                gpCreateInfo.outputState.renderTargetCount      = 1;
-                gpCreateInfo.outputState.renderTargetFormats[0] = GetSwapchain()->GetColorFormat();
-                gpCreateInfo.outputState.depthStencilFormat     = GetSwapchain()->GetDepthFormat();
-                gpCreateInfo.pPipelineInterface                 = mSphere.pipelineInterface;
-                PPX_CHECKED_CALL(GetDevice()->CreateGraphicsPipeline(&gpCreateInfo, &mPipelines[pipelineIndex++]));
-
-                // Position Planar Pipeline
-                gpCreateInfo.vertexInputState.bindingCount = 2;
-                gpCreateInfo.vertexInputState.bindings[0]  = mSphereMeshes[2 * k + 1]->GetDerivedVertexBindings()[0];
-                gpCreateInfo.vertexInputState.bindings[1]  = mSphereMeshes[2 * k + 1]->GetDerivedVertexBindings()[1];
-                PPX_CHECKED_CALL(GetDevice()->CreateGraphicsPipeline(&gpCreateInfo, &mPipelines[pipelineIndex++]));
-            }
-        }
+    // Pre-load the current pipeline variant.
+    GetSpherePipeline();
+}
+
+Result GraphicsBenchmarkApp::CompileSpherePipeline(const SpherePipelineKey& key)
+{
+    if (mPipelines.find(key) != mPipelines.end()) {
+        return SUCCESS;
+    }
+    grfx::GraphicsPipelinePtr pipeline    = nullptr;
+    bool                      interleaved = key.vertexAttributeLayout == 0;
+    size_t                    meshIndex   = 2 * key.vertexFormat + (interleaved ? 0 : 1);
+    grfx::BlendMode           blendMode   = (key.enableAlphaBlend ? grfx::BLEND_MODE_ALPHA : grfx::BLEND_MODE_NONE);
+    // Interleaved pipeline
```

Comment:
nit: Was this comment supposed to be inside the if block below?

Reply:
Done.